routers/api/v1/repo/file.go

```diff
@@ -597,3 +599,86 @@ func GetContentsList(ctx *context.APIContext) {
 	// same as GetContents(), this function is here because swagger fails if path is empty in GetContents() interface
 	GetContents(ctx)
 }
+
+// GetFileHistory get a file's commit history
+func GetFileHistory(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/git/history/{filepath} repository repoGetFileHistory
+	// ---
+	// summary: Get a file's or directory's commit history
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: filepath
+	//   in: path
+	//   description: filepath of the file/directory
+	//   type: string
+	//   required: true
+	// - name: ref
+	//   in: query
+	//   description: "The name of the branch. Default the repository’s default branch (usually master)"
+	//   type: string
+	//   required: false
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/CommitList"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+
+	if ctx.Repo.Repository.IsEmpty {
+		ctx.NotFound()
+		return
+	}
+
+	ref := ctx.FormTrim("ref")
+	if len(ref) < 1 {
+		ref = ctx.Repo.Repository.DefaultBranch
+	}
+
+	page := ctx.FormInt("page")
+	if page <= 1 {
+		page = 1
+	}
+
+	commitsCount, err := ctx.Repo.GitRepo.FileCommitsCount(ref, ctx.Repo.TreePath)
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "FileCommitsCount", err)
+		return
+	} else if commitsCount == 0 {
```

Comment:
Will the commitCount be 0 when the file is not present?
Or will it return an error?
Because I don't think we should return a `50X` if the file is not present…

Reply:
Yes, I think so (didn't test, I'll have to do this, but the web ui does the same)

Reply:
Yes, Gitea returns a 404 if file does not exist

Reply:
Ok, you're right, it returns a 500 error if the file does not exist, I'll update this.

Reply:
After testing again, I noticed that it returns a 404 error if the file doesn't exist and never existed (my previous tests were wrong)